polygeist/lib/Dialect/Polygeist/Transforms/ArgumentPromotion.cpp

```diff
@@ -216,7 +216,7 @@ class ArgumentPromotionPass
 
   /// Return true is the callee is a candidate, and false otherwise.
   bool isCandidateCallable(CallableOpInterface callableOp,
-                           const polygeist::FunctionKernelInfo &funcKernelInfo);
+                           polygeist::FunctionKernelInfo &funcKernelInfo);
```

Comment:
can we add back these `const` in the 3 passes?

Reply:
I further simplified the interface making functions `static`. Now we don't even need to build the `FunctionKernelInfo` in most cases.